sw/device/lib/dif/dif_otbn.c

```diff
@@ -201,33 +217,28 @@ dif_otbn_result_t dif_otbn_is_busy(const dif_otbn_t *otbn, bool *busy) {
   return kDifOtbnOk;
 }
 
-dif_otbn_result_t dif_otbn_get_err_code(const dif_otbn_t *otbn,
-                                        dif_otbn_err_code_t *err_code) {
-  if (otbn == NULL || err_code == NULL) {
+static const uint32_t kDifOtbnErrBitsAll =
+    (1 << kDifOtbnErrBitBadDataAddr) | (1 << kDifOtbnErrBitBadInsnAddr) |
+    (1 << kDifOtbnErrBitCallStack) | (1 << kDifOtbnErrBitIllegalInsn) |
+    (1 << kDifOtbnErrBitLoop) | (1 << kDifOtbnErrBitFatalImem) |
+    (1 << kDifOtbnErrBitFatalDmem) | (1 << kDifOtbnErrBitFatalReg);
+
+dif_otbn_result_t dif_otbn_get_err_bits(const dif_otbn_t *otbn,
+                                        uint32_t *err_bits) {
+  if (otbn == NULL || err_bits == NULL) {
     return kDifOtbnBadArg;
   }
 
-  uint32_t err_code_raw =
-      mmio_region_read32(otbn->base_addr, OTBN_ERR_CODE_REG_OFFSET);
-
-  // Ensure that all values returned from hardware match explicitly defined
-  // values in the DIF.
-  switch (err_code_raw) {
-    case kDifOtbnErrCodeNoError:
-    case kDifOtbnErrCodeBadDataAddr:
-    case kDifOtbnErrCodeBadInsnAddr:
-    case kDifOtbnErrCodeCallStack:
-    case kDifOtbnErrCodeIllegalInsn:
-    case kDifOtbnErrCodeLoop:
-    case kDifOtbnErrCodeFatalImem:
-    case kDifOtbnErrCodeFatalDmem:
-    case kDifOtbnErrCodeFatalReg:
-      *err_code = (dif_otbn_err_code_t)err_code_raw;
-      return kDifOtbnOk;
+  uint32_t err_bits_raw =
+      mmio_region_read32(otbn->base_addr, OTBN_ERR_BITS_REG_OFFSET);
 
-    default:
-      return kDifOtbnUnexpectedData;
+  // Check the error bits from hardware all correspond to defined errors.
+  if ((err_bits_raw & ~kDifOtbnErrBitsAll) != 0) {
+    return kDifOtbnUnexpectedData;
   }
```

Comment:
Is this actually necessary? No other DIF bothers to check that the hardware has bothered to include extra bits.

Reply:
I sort of agree with @mcy here. I suppose we might worry about software getting confused with stray bits that got set (maybe if someone is glitching the register read). Maybe we should just be setting `*err_bits = err_bits_raw & kDifOtbnErrBitsAll`?

Reply:
I'm happy to drop it, I was simply aiming to match the existing behaviour (which returns `kDifOtbnUnexpectedData` if the error code wasn't one of the defined ones).

@imphil thoughts?